python/equistore-core/equistore/core/labels.py

```diff
@@ -109,134 +240,208 @@ def __new__(cls, names: Union[Sequence[str], str], values: np.ndarray, **kwargs)
         except TypeError as e:
             raise TypeError("Labels values must be convertible to integers") from e
 
-        dtype = [(name, np.int32) for name in names]
+        self._lib = _get_library()
+        self._labels = _create_new_labels(self._lib, names, values)
+        self._names = names
+        self._values = _labels_values(self._labels)
 
-        if values.shape[1] != 0:
-            values = values.view(dtype=dtype).reshape((values.shape[0],))
+    @staticmethod
+    def single() -> "Labels":
+        """
+        Create :py:class:`Labels` to use when there is no relevant metadata and
+        only one entry in the corresponding dimension (e.g. keys when a tensor
+        map contains a single block).
+        """
+        return Labels(names=["_"], values=np.zeros(shape=(1, 1), dtype=np.int32))
 
-        obj = values.view(cls)
+    @staticmethod
+    def empty(names: Union[str, Sequence[str]]) -> "Labels":
+        """
+        Create :py:class:`Labels` with given ``names`` but no values.
 
+        :param names: names of the dimensions in the new labels. A single string
+                      is transformed into a list with one element, i.e.
+                      ``names="a"`` is the same as ``names=["a"]``.
+        """
+        return Labels(names=names, values=np.array([]))
+
+    @staticmethod
+    def range(name: str, end: int) -> "Labels":
+        """
+        Create :py:class:`Labels` with a single dimension using the given
+        ``name`` and values in the ``[0, end)`` range.
+
+        :param name: name of the single dimension in the new labels.
+        :param end: end of the range for labels
+
+        >>> from equistore import Labels
+        >>> labels = Labels.range("dummy", 7)
+        >>> labels.names
+        ['dummy']
+        >>> print(labels.values)
+        [[0]
+         [1]
+         [2]
+         [3]
+         [4]
+         [5]
+         [6]]
+        """
+        return Labels(
+            names=[name],
+            values=np.arange(end, dtype=np.int32).reshape(-1, 1),
```

Comment:
because TorchScript! We used to do that but to provide a nice API we would need support for `*args, **kwargs` (to handle all of `arange(3)`, `arange(stop=3)`, `arange(3, 5)`, `arange(stop=5, start=3)`); but TorchScript does not support any of that =/